pkg/planner/cardinality/ndv.go

```diff
@@ -16,16 +16,25 @@ package cardinality
 
 import (
 	"math"
+	"slices"
 
 	"github.com/pingcap/tidb/pkg/expression"
 	"github.com/pingcap/tidb/pkg/planner/property"
+	"github.com/pingcap/tidb/pkg/sessionctx/vardef"
+	"github.com/pingcap/tidb/pkg/sessionctx/variable"
 	"github.com/pingcap/tidb/pkg/statistics"
 	"github.com/pingcap/tidb/pkg/util/logutil"
 	"go.uber.org/zap"
 )
 
 const distinctFactor = 0.8
 
+// SessionVarProvider provides access to session variables for NDV estimation.
+// Both sessionctx.Context and planctx.PlanContext implement this interface.
+type SessionVarProvider interface {
```

Comment:
Can we just use `ctx planctx.PlanContext` directly? Since `cardinality` is a sub-package of the planner, using `PlanCtx` directly seems clearer and simpler here, and all other cardinality functions are also using `PlanCtx` directly like `Selectivity()`.